e2e/api_test.go

```diff
@@ -425,6 +441,180 @@ func TestE2E_TaskEndpoints_Delete_Conflict(t *testing.T) {
+	resourcesPath := filepath.Join(tempDir, taskName, resourcesDir)
+	validateServices(t, false, []string{service.ID}, resourcesPath)
+
+	// Make a change that triggers the new task, expect an event
+	service = testutil.TestService{
+		ID:      serviceName + "-2",
+		Name:    serviceName,
+		Address: "5.6.7.9",
+		Port:    8080,
+	}
+	testutils.RegisterConsulService(t, srv, service, defaultWaitForRegistration)
+	time.Sleep(defaultWaitForNoEvent)
+	resp = testutils.RequestHTTP(t, http.MethodGet, s, "")
+	assert.Equal(t, http.StatusOK, resp.StatusCode)
+	resourcesPath = filepath.Join(tempDir, taskName, resourcesDir)
+	validateServices(t, true, []string{service.ID}, resourcesPath)
+}
+
+// TestE2E_TaskEndpoints_Create_Run_Now tests the create task endpoint with run now.
+// This runs a Consul server and the CTS binary in daemon mode.
+// POST /v1/tasks
+func TestE2E_TaskEndpoints_Create_Run_Now(t *testing.T) {
+	t.Parallel()
+	// Test creating a task
+	// 1. Start with a task
+	// 2. Create infrastructure change that would trigger new task
+	// 3. Create a new task with run=now
+	// 4. Check that the new task exists
+	// 5. Check that new task ran based on (2) infrastructure change
+
+	srv := testutils.NewTestConsulServer(t, testutils.TestConsulServerConfig{
+		HTTPSRelPath: "../testutils",
+	})
+	defer srv.Stop()
+	tempDir := fmt.Sprintf("%s%s", tempDirPrefix, "create_task")
+	initialTaskName := "initial-task"
+	cts := ctsSetup(t, srv, tempDir,
+		moduleTaskConfig(initialTaskName, "./test_modules/local_instances_file"))
+
+	// Make a change that would trigger the new task, if it existed
+	taskName := "created-task"
+	serviceName := "testService"
+	service := testutil.TestService{
+		ID:      serviceName,
+		Name:    serviceName,
+		Address: "5.6.7.8",
+		Port:    8080,
+	}
+	testutils.RegisterConsulService(t, srv, service, defaultWaitForRegistration)
+
+	// Create task
+	u := fmt.Sprintf("http://localhost:%d/v1/tasks",
+		cts.Port())
+	u = fmt.Sprintf("%s?run=now", u)
+
+	createTaskRequest := fmt.Sprintf(createTestTaskTemplate, taskName, serviceName)
+
+	resp := testutils.RequestHTTP(t, http.MethodPost, u, createTaskRequest)
+	bodyBytes, err := io.ReadAll(resp.Body)
+	require.NoError(t, err)
+	require.Equal(t, http.StatusOK, resp.StatusCode, "response body %s", string(bodyBytes))
+
+	// Check that the task has been created
+	s := fmt.Sprintf("http://localhost:%d/%s/status/tasks/%s",
+		cts.Port(), "v1", taskName)
+	resp = testutils.RequestHTTP(t, http.MethodGet, s, "")
+	assert.Equal(t, http.StatusOK, resp.StatusCode)
+
+	// Verify that there was an update
+	resourcesPath := filepath.Join(tempDir, taskName, resourcesDir)
+	validateServices(t, true, []string{service.ID}, resourcesPath)
+}
+
+// TestE2E_TaskEndpoints_InvalidSchema tests the create task endpoint with an invalid schema, no task
+// should be created. This runs a Consul server and the CTS binary in daemon mode.
+//	POST /v1/tasks
+func TestE2E_TaskEndpoints_InvalidSchema(t *testing.T) {
```

Comment:
it looks like this test is mainly exercising the code that parses the request and checking that a task isn't created. could this possibly be done as a unit/integration test?

Reply:
The invalid schema is checked by the generated code which requires setting up the routing, that's why I included it as an integration test

Reply:
To clarify, you mean that it is why you made it an e2e test? Or do you mean that it can be an integration test?

Reply:
Sorry, yes, this could potentially be an integration test, my experience with e2e and integration tests in the past were that they could be synonymous (which they aren't here) so I was confused. I'll look into potentially making this an integration test.

Reply:
Gotcha. Yea, within CTS repo, we have some integration tests that aren't in our e2e directory and only exercise a targeted piece of code rather than running the binary. If the integration test isn't feasible or doesn't have less code/complexity, then don't worry about it

Reply:
Thinking about this some more, to get this working as an integration test would require I setup the routing and handling in the test itself. Since the test is really to make sure that everything is setup properly (including the middleware that would reject this invalid request) the integration test wouldn't really verify that. It would just be testing that the generated code does what it's supposed to do, which isn't what my goal was. I'm going to leave this as an e2e test.